mne/preprocessing/xdawn.py

```diff
@@ -129,31 +126,143 @@ def least_square_evoked(epochs, return_toeplitz=False):
         If return_toeplitz is true, return the toeplitz matrix for each event
         type in epochs.event_id.
     """
-    if not isinstance(epochs, _BaseEpochs):
-        raise ValueError('epochs must be an instance of `mne.Epochs`')
-
-    events = epochs.events.copy()
-    events[:, 0] -= events[0, 0] + int(epochs.tmin * epochs.info['sfreq'])
-    data = _construct_signal_from_epochs(epochs)
-    evoked_data, toeplitz = _least_square_evoked(data, events, epochs.event_id,
-                                                 tmin=epochs.tmin,
-                                                 tmax=epochs.tmax,
-                                                 sfreq=epochs.info['sfreq'])
+
+    events = events.copy()
+    events[:, 0] -= events[0, 0] + int(tmin * info['sfreq'])
+    data = _construct_signal_from_epochs(events, info['sfreq'], tmin, tmax,
+                                         epochs_data)
+    evoked_data, toeplitz = _least_square_evoked(data, events, event_id,
+                                                 tmin=tmin, tmax=tmax,
+                                                 sfreq=info['sfreq'])
     evokeds = dict()
-    info = cp.deepcopy(epochs.info)
+    info = cp.deepcopy(info)
     for name, data in evoked_data.items():
-        n_events = len(events[events[:, 2] == epochs.event_id[name]])
-        evoked = EvokedArray(data, info, tmin=epochs.tmin,
+        n_events = len(events[events[:, 2] == event_id[name]])
+        evoked = EvokedArray(data, info, tmin=tmin,
                              comment=name, nave=n_events)
-        evokeds[name] = evoked
+        evokeds[name] = evoked.data
```

Comment:
If you pass evoked.data why do we need to create an EvokedArray. Can t we just stay with np arrays ? In which case info is also not needed ?